### PR TITLE
[ticket/15836] Event to send message via external transport

### PR DESCRIPTION
**Pull request by xchwarze**
Checklist:

- [x] Correct branch: master for new features; 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15836

### CODE REVIEW

File: phpBB/includes/functions_messenger.php [hunk 1]
```diff
@@ -545,6 +545,33 @@ function msg_email()
...
+		* @var	string	msg			The message text
+		* @since 3.2.2
+		*/
+		$vars = array(
+			'break',
+			'addresses',
+			'subject',
+			'msg',
+		);
+		extract($phpbb_dispatcher->trigger_event('core.notification_message_email', compact($vars)));
```
Comment (rubencm) [inline phpBB/includes/functions_messenger.php]: Hello, i think you should set variables back in case the event modifies them

Comment (xchwarze) [reply phpBB/includes/functions_messenger.php]: Hello! for change the content of subject and msg alredy exist core.modify_notification_message

This new hook is for use with mailgun/others mails transport

File: phpBB/includes/functions_messenger.php [hunk 2]
```diff
@@ -545,6 +545,33 @@ function msg_email()
...
+		$msg = $this->msg;
+		/**
+		* Event to send message via external transport
+		*
+		* @event core.notification_message_email
+		* @var	bool	break		Flag indicating if the function return after hook
+		* @var	array	addresses 	The message recipients
+		* @var	string	subject		The message subject
+		* @var	string	msg			The message text
+		* @since 3.2.2
```
Comment (marc1706) [inline phpBB/includes/functions_messenger.php]: Please update this to 3.2.4-RC1

File: phpBB/includes/functions_messenger.php [hunk 3]
```diff
@@ -878,23 +905,45 @@ function process()
...
+						$break = false;
+						/**
+						* Event to send message via external transport
+						*
+						* @event core.notification_message_email
+						* @var	bool	break		Flag indicating if the function return after hook
+						* @var	array	addresses 	The message recipients
+						* @var	string	subject		The message subject
+						* @var	string	msg			The message text
+						* @since 3.2.2
```
Comment (marc1706) [inline phpBB/includes/functions_messenger.php]: Please update this to 3.2.4-RC1

**Comment by marc1706** [comment]
Rebased, closed, and reopened against 3.2.x.

**Comment by marc1706** [comment]
On a side note, please try to follow our commit message format in future commits: https://area51.phpbb.com/docs/dev/3.2.x/development/git.html#commit-messages

Thanks.

**Pull request merged**